src/plugins/stylesheet/PostCSSPlugin.ts

```diff
@@ -58,34 +58,29 @@ export class PostCSSPluginClass implements Plugin {
         file.bustCSSCache = true;
         file.loadContents();
 
-        let paths: string[] = this.options && this.options.paths || [];
-        paths.push(file.info.absDir);
+        let generateSourceMaps = true;
+        let postCssOptions;
+        let pathOptions: string[] = [file.info.absDir];
+
+        if (this.options) {
+            const {sourceMaps = generateSourceMaps, plugins, paths = pathOptions, ...otherOptions} = this.options;
+            generateSourceMaps = sourceMaps;
+            postCssOptions = otherOptions;
+            pathOptions = paths;
+        }
 
         const cssDependencies = file.context.extractCSSDependencies(file, {
-            paths: this.options && this.options.paths || [file.info.absDir],
+            paths: pathOptions,
             content: file.contents,
             extensions: ["css"]
-        })
+        });
 
         if (!postcss) {
             postcss = require("postcss");
         }
-        let generateSourceMaps = true;
-        let postCSSPlugins = [];
-        if (Array.isArray(this.options)) {
-            postCSSPlugins = this.options;
-        } else {
-            if (this.options) {
-                if (Array.isArray(this.options.plugins)) {
```

Comment:
Can we still keep it compatible with the old version? I believe some people are relying on it. Let's just update the doc to the latest

Reply:
Ok but the API will look weird:

There is two options to declare postcss plugins:
```js
[PostCSS(plugins)]
```

and : 

```js
[PostCSS([], {plugins: plugins})]
```

What happens if both are defined? Should we merge them?

Reply:
I think it was supposed to look like:

```
[PostCSS({plugins : []})]
```
Like here http://fuse-box.org/plugins/postcss-plugin#paths-for-hmr

Reply:
I think it's either the proper way (you suggested) or the old one, but we need to clean up the docs

Reply:
Ok I set it up to merge them, this ensures maximum compatibility right now.
This way, we can defer this decision if you want.

I like you suggestion but it would be a breaking change.
Your call 😄

Reply:
Why a breaking change? That's how it is working now
either
`[PostCSS({plugins : []})]` or `[PostCSS([]]`

WE just need to check for the first argument and check if it's an object with `plugins` property how it was before

Reply:
Unfortunately, I think it is broken right now.
According to the postcss docs, we can only pass the plugins (what we refer in the code as processors) as the first argument of the postcss function.

What fusebox [does today](https://github.com/fuse-box/fuse-box/blob/8134ac6d151daf98d409f09389562dc49ecc7e44/src/plugins/stylesheet/PostCSSPlugin.ts#L88) does not work.

As you can see the [processOptions](http://api.postcss.org/global.html#processOptions) (second argument to `postcss().process()`) does not take plugins.

But I'll make it work as you suggested, no problem.

Reply:
Oh, that's a bummer ( Can you update the docs please?

Reply:
Hm, i think you got me wrong here ;-( I liked your suggestion - to align with postcss style, as in
```js
PostCSS([plugin1, plugin2], {  /options/ })
```

And I think it should be like that... It's just wanted to keep the old style and update the documentation (removing the docs on the incorrect one).

To avoid breaking changes, you know?

Reply:
No problem @nchanged 😃 
I'll go back to concatenating both "processors" and the plugins option then, right ?

Reply:
like in PostCSS yes, except make an additional condition for the first argument, if it's an object, take `plugins` and the pass the rest to postcss options